### PR TITLE
LEGLINK-839: Automation UI : Normalization Sequence allows duplicate operations to be added

### PR DESCRIPTION
**Pull request by SaraMunich-Lantana**
### 🛠️ Description of Changes

Updated the Normalization Sequence configuration to prevent the same normalization operation from being added multiple times to a sequence. Operations already selected within the sequence are disabled in other operation dropdowns, available options are refreshed when entries are changed or removed, and the Add Operation button is disabled when all available operations have been used. Added server-side validation to reject sequence save requests containing duplicate operation IDs.

### 🧪 Testing Performed

Built and ran the Automation UI locally and verified the Normalization Sequence configuration behavior. Confirmed that operations already selected in a sequence cannot be selected again, removing an operation makes it available again, and the Add Operation button is disabled when all available operations have been added and re-enabled when an operation is removed. Saved and reopened a sequence to verify the configured operations and ordering persisted correctly. Also manually submitted a sequence containing duplicate operation IDs and confirmed the server returned HTTP 400 with the expected duplicate-operation validation message.

### 🧑‍🔬 Unit Testing

- Coverage: 66.7%

N/A

### 📓 Documentation Updated

N/A

### CODE REVIEW

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/ServiceTestSuiteBase.cs [hunk 1]
```diff
@@ -303,8 +303,7 @@ private static string Truncate(string? value, int maxLength = 300) =>
 
     private static string BodyOrNote(string? value, string note)
     {
-        var body = Truncate(value);
-        return string.IsNullOrWhiteSpace(body) ? note : body;
+        return string.IsNullOrWhiteSpace(value) ? note : value;
```
Comment (nvmLantana) [inline DotNet/Automation.UI/Services/ApiHealth/TestSuites/ServiceTestSuiteBase.cs]: This is the LEGLINK-904 / #1863 change (`BodyOrNote` no longer truncates). Doesn't belong on this PR. Pull the API Health files out so we review 904 over there.

File: DotNet/Automation.UI/Views/Normalizations/Index.cshtml [hunk 2]
```diff
@@ -865,14 +869,67 @@
...
+        const selectedOperationIds = new Set(
+            Array.from(container.querySelectorAll('.seq-op-select'))
+                .map(select => select.value).filter(Boolean)
+        );
+
+        const selectedOperationId = entry?.operationId
+            || allOperations.find(o => !selectedOperationIds.has(o.id))?.id
+            || '';
+
+        if (!selectedOperationId) {
```
Comment (nvmLantana) [inline DotNet/Automation.UI/Views/Normalizations/Index.cshtml]: Silent `return` when every operation is already in the sequence. `#btnAddSeqEntry` gets disabled, so this is mostly belt-and-suspenders, but if Add still fires the user gets no warning. Ticket asked for a warning.

Uniqueness is by operation entity id, not type. Two different CopyProperty configs can still both be in a sequence. That's probably what we want, just calling it out.

File: DotNet/Automation.UI/Controllers/NormalizationsController.cs [hunk 3]
```diff
@@ -129,6 +129,11 @@ public async Task<IActionResult> SaveSequence([FromBody] NormalizationSequenceDe
         if (model.Entries.Count == 0)
             return BadRequest("At least one operation entry is required.");
 
+        if (model.Entries.GroupBy(e => e.OperationId).Any(g => g.Count() > 1))
```
Comment (nvmLantana) [inline DotNet/Automation.UI/Controllers/NormalizationsController.cs]: This is the right check for 839. No test covering SaveSequence rejecting duplicate OperationIds. Coverage on this PR is 6.2% / N/A. A controller test that posts two entries with the same OperationId and asserts 400 would make this actually proven.

**Review by nvmLantana — CHANGES_REQUESTED** [review]
This PR is titled LEGLINK-839, but most of the diff isn't that ticket.

`cfb33914` is the API Health payload-truncation change from LEGLINK-904 / https://github.com/lantanagroup/link-cloud/pull/1863 (same 13 files). Drop those from this branch so 839 stays scoped to the sequence editor. 904 already has its own PR.

What's here for 839 looks right in spirit (client disable + server `GroupBy(OperationId)` reject). A couple nits on the 839 code in the inline comments. Also no test that SaveSequence actually returns 400 on duplicates.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/ServiceTestSuiteBase.cs [hunk 4]
```diff
@@ -303,8 +303,7 @@ private static string Truncate(string? value, int maxLength = 300) =>
...
     }
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/ServiceTestSuiteBase.cs]: Good point regarding the Mongo document-size risk. The removal of the existing truncation is intentional for LEGLINK-904 because the previous 300/500-character limits prevent API Health users from inspecting complete request/response payloads. I’m reviewing the API Health persistence path to determine whether a separate defensive persistence-size limit is needed without reintroducing the user-visible truncation this ticket is addressing.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/ValidationServiceTestSuite.cs [hunk 5]
```diff
@@ -202,9 +202,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/ValidationServiceTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Validation service response payload. Reintroducing the previous limit would restore the behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/TenantServiceTestSuite.cs [hunk 6]
```diff
@@ -429,9 +429,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/TenantServiceTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Tenant service response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/SubmissionServiceTestSuite.cs [hunk 7]
```diff
@@ -202,9 +202,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/SubmissionServiceTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Submission service response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/ReportServiceTestSuite.cs [hunk 8]
```diff
@@ -427,9 +427,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/ReportServiceTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Report service response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffTestSuite.cs [hunk 9]
```diff
@@ -248,7 +248,7 @@ private async Task<ApiTestRunResult> CallBffAsync(
             var body = response.RawBody;
             var capturedBody = string.IsNullOrWhiteSpace(body)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : (body.Length > 500 ? body[..500] : body);
+                : body;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Admin BFF response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffTestSuite.cs [hunk 10]
```diff
@@ -398,9 +398,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete AdminBFF service response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffAuthTestSuite.cs [hunk 11]
```diff
@@ -206,7 +206,7 @@ private async Task<ApiTestRunResult> CallEndpointAsync(
             result.Passed = responseCodeMatchesExpected;
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {(int)response.StatusCode})."
-                : (responseBody.Length > 500 ? responseBody[..500] : responseBody);
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/AdminBffAuthTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete AdminBFF service response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Models/ApiHealth/ApiTestRunResult.cs [hunk 12]
```diff
@@ -54,6 +54,6 @@ public sealed class ApiTestRunResult
     /// <summary>Trace ID from the response headers for correlating with Grafana/distributed tracing.</summary>
     public string? TraceId { get; set; }
 
-    /// <summary>Truncated response body (for both success and failure diagnostics).</summary>
+    /// <summary>Response body (for both success and failure diagnostics).</summary>
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Models/ApiHealth/ApiTestRunResult.cs]: The documentation change is intentional for LEGLINK-904. ResponseBody is no longer truncated at the API Health result capture points, so describing this property as a truncated response body would no longer accurately reflect its current behavior. The separate concern regarding whether a defensive size limit should be applied at the persistence layer is being reviewed independently.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/CensusServiceTestSuite.cs [hunk 13]
```diff
@@ -406,9 +406,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/CensusServiceTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Census response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/DataAcquisitionTestSuite.cs [hunk 14]
```diff
@@ -579,9 +579,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/DataAcquisitionTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete DataAcquisition response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/MeasureEvalTestSuite.cs [hunk 15]
```diff
@@ -223,9 +223,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/MeasureEvalTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete MeasureEval response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/NormalizationTestSuite.cs [hunk 16]
```diff
@@ -329,9 +329,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/NormalizationTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete Normalization response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Services/ApiHealth/TestSuites/QueryDispatchTestSuite.cs [hunk 17]
```diff
@@ -332,9 +332,7 @@ private async Task<ApiTestRunResult> CallRawGetAsync(
 
             result.ResponseBody = string.IsNullOrWhiteSpace(responseBody)
                 ? $"No response body was returned (HTTP {result.ActualStatusCode})."
-                : responseBody.Length > 500
-                    ? responseBody[..500]
-                    : responseBody;
+                : responseBody;
```
[the remark this answers is not in the document]
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Services/ApiHealth/TestSuites/QueryDispatchTestSuite.cs]: The removal of the 500-character truncation here is intentional for LEGLINK-904 so API Health users can inspect the complete QueryDispatch response payload. Reintroducing the previous limit would restore the truncation behavior this ticket is intended to fix. I’m reviewing the API Health persistence path separately to determine whether a defensive storage-size limit or other safeguard is appropriate at the persistence layer without truncating normal user-visible response content.

File: DotNet/Automation.UI/Controllers/NormalizationsController.cs (hunk 3 above)
Comment (SaraMunich-Lantana) [reply DotNet/Automation.UI/Controllers/NormalizationsController.cs]: Added a new NormalizationsControllerTests.cs unit test

**Review by nvmLantana — APPROVED** [review]
